crates/forge/tests/integration/syscalls.rs

```diff
@@ -142,3 +142,29 @@ fn library_call_syscall() {
 
     assert_passed!(result);
 }
+
+#[test]
+fn test_call_syscall_fail_in_test_fn() {
+    let test = test_case!(indoc!(
+        r#"
+        use starknet::{ get_block_timestamp };
+        #[test]
+        fn test_execute_disallowed_syscall() {
+            get_block_timestamp();
+        }
+    "#
+    ));
+
+    let result = run(
+        &test.path().unwrap(),
+        &test.path().unwrap().join("src/lib.cairo"),
+        &Some(test.linked_libraries()),
+        &Default::default(),
+        Some(&Utf8PathBuf::from_path_buf(corelib().to_path_buf()).unwrap()),
+        &test.contracts(corelib().path()).unwrap(),
+        &Utf8PathBuf::from_path_buf(predeployed_contracts().to_path_buf()).unwrap(),
+    )
+    .unwrap();
+
+    assert_failed!(result);
```

Comment:
We could also assert that the output contains "starknet syscalls cannot be used in tests", we have a macro `assert_case_output_contains!` for that

Reply:
I'd definitely use that